### ISSUE TITLE
Display highways with access=no/private grayed out

### ISSUE THREAD

**Issue by map-per** [organicmaps/organicmaps#1454]
It would be good to display roads and paths that are tagged with access=no or private grayed out to allow mapusers to avoid such paths when they plan their rout themselves. 
E.g. similar to how it is done in OSM Carto (https://www.openstreetmap.org/#map=18/48.18987/11.62815):
![Screenshot (723)](https://user-images.githubusercontent.com/79519062/138491661-c5cf84c9-e79f-4e40-9904-f173f2a81b50.png)

**Comment by matkoniecz** [comment]
Or marked in some other way. Currently it is quite significant limitation affecting many places where part of paths/roads has limited access.

Though note `access=no foot=designated highway=path` tagging!

**Comment by bfontaine** [comment]
See also #1031 and maybe #979.

**Comment by bfontaine** [comment]
I can confirm this issue is still relevant.

See for example [this way](https://www.openstreetmap.org/way/238651628) on OSM carto: https://www.openstreetmap.org/#map=19/40.72594/13.95874
![OSM Carto](https://github.com/organicmaps/organicmaps/assets/1334295/d67dd390-47b7-4825-8f76-90a3b93c4fd5)

The same on Organic Maps:
![Organic Maps](https://github.com/organicmaps/organicmaps/assets/1334295/ba765e17-1cb4-40d6-b718-32ddc74a1e32)

Worse, it routes through it:
![Organic Maps routing](https://github.com/organicmaps/organicmaps/assets/1334295/89277be9-4153-4919-a237-bffbfe30762f)

Versions:
- iPhone13,1 (16.5)
- Organic Maps 2023.05.08-1
- fr-FR
- fr-FR, en-FR, it-FR

**Comment by vng** [comment]
- Snapping start/end points on a private road is a bug.
- @pastk Can wi try draw hwtag=private the same as in OSM? eval(width) in styles should help here to define in nicely

**Comment by pastk** [comment]
> 
>     * @pastk Can wi try draw hwtag=private the same as in OSM? eval(width) in styles should help here to define in nicely

Could you please elaborate what is your idea of using `eval(width)` for this case? How will it help?

**Comment by vng** [comment]
Draw a dashed gray line inside a parent highway with width=eval(width) * 0.7
Thus we can define one hwtag=private rule for all highway types.

**Comment by pastk** [comment]
But evals never worked like this in kothic..
You want to create all possible `highway-*-private` combinations types and make them take the width property from the "base" type?

**Comment by vng** [comment]
AFAIR it worked before. Like this, somewhere in the end after defining all highways:
```
z17-[highway][hwtag=private]::private_dashes
{ color: grey; dashes: 5, 5; width: eval(prop("width")*0.7); }
```

**Comment by vng** [comment]
One more case when dynamic eval id good:
```
node|z15-[layer < 0]
{ text-opacity: 0.5; symbol-opacity: 0.5 }
```

**Comment by biodranik** [comment]
@pastk the goal is not to make an universally configurable styling mechanism, but to make _the fastest_ styling mechanism, easy to support. 

If hard-coding something is way better than doing in in the style, then definitely hard-code it and document it!

**Comment by vng** [comment]
@pastk You are right here! Code + simple color/opacity config is much better for this problem, rather than inventing complicated styles pipeline.

**Comment by map-per** [comment]
> Though note `access=no foot=designated highway=path` tagging!

I thing this approach described by imagico is good: https://imagico.de/blog/en/rethinking-road-access-restrictions-rendering/
It basicly suggests to distinct when a `highway` is considered private/not usable by `highway` type. A foodway would e.g. be private if `foot` is not allowed.

**Comment by pastk** [comment]
In the routing mode it'll help to visually distinguish non-accessible roads based on the current mode of transport (car, bicycle, prob we should add motorcycle and trucks at some point..).

**Comment by map-per** [comment]
> I thing this approach described by imagico is good: https://imagico.de/blog/en/rethinking-road-access-restrictions-rendering/

This approach is now implemented in OSMCarto and live on openstreetmap.org. This [blogpost](https://www.openstreetmap.org/user/imagico/diary/405334) describes the implementation in more detail.

**Comment by mythsunwind** [comment]
I recently experienced a problem with a path with `access=no` in a nature reserve in Germany. It is not allowed to leave paths in such areas and you can be fined for it. The path does not look any different on the map. Even when tapping on it there is no information that this path is not allowed to be used.

**Comment by StC-OSM** [comment]
> I recently experienced a problem with a path with `access=no` in a nature reserve in Germany. It is not allowed to leave paths in such areas and you can be fined for it. The path does not look any different on the map. Even when tapping on it there is no information that this path is not allowed to be used.

... and we are having difficult discussions in France with land owners who ask us to remove ways from OSM because "access=no" and "access=private" are not properly handled in some applications. In the US, the Trails Working Group has managed to obtain that major renderings and apps are well-behaved, thus allowing to reach an agreement between contributors and land managers. It would be much appreciated if OrganicMaps could join that club of well-behaved apps.

**Comment by Jean-BaptisteC** [comment]
I have asked opinion to the french community https://t.me/OrganicMapsFR/993 and the most of people do not want we hide private paths from the maps.
IMO, we can reduce visibility of private path by reduce visibility on low level and improve styles (maybe dashed paths or add add transparency?)

**Comment by stchatty** [comment]
What about adding red crosses on top of private paths, indicating that they should not be used?

**Comment by StC-OSM** [comment]
What about adding red crosses on top of private paths, do indicate that they should not be used?
